analyzers/ImpactParameter.py

```diff
@@ -1,48 +1,214 @@
 from heppy.framework.analyzer import Analyzer
-from ROOT import TVector3
-#from heppy.framework.event import Event
+from ROOT import TFile, TH1F
+from ROOT import TVector3, TLorentzVector
 from heppy.papas.path import Helix
-#import collections
+import math
+from heppy.utils.deltar import deltaR
 
 class ImpactParameter(Analyzer):
-
+    '''This analyzer puts an impact parameter for every charged particle
+    as an attribute of its path.
+    The significance is calculated, the calculus are a first order approximation,
+    thus this may not be correct for large impact parameters (more than 3 mm).
+    The Impact parameter significance is stored in the particle's path.
+    
+    New attributes for particles.path (from compute_IP) :
+    *   closest_t = time of closest approach to the primary vertex.
+    *   IP = signed impact parameter
+    *   IPx, IPy, IPz = coordinates of the point of closest approach to the
+        primary vertex, with respets to the primary vertex
+        
+    New attributes for particles.path (from compute_theta_0) :
```

Comment:
actually for jets, I see that you are using jet.tags to store your taggers. this is very good. But you can still use some kind of class for additional jet-wide tagging information. 
for instance: jet.tags['TCHE'] should contain the TCHE value. 
if you want to store more in the jet, you need to add a BTagInfo